pyro/contrib/gp/models/svgp.py

```diff
@@ -146,7 +151,7 @@ def guide(self):
             pyro.sample(u_name,
                         dist.MultivariateNormal(u_loc, scale_tril=u_scale_tril)
                             .independent(u_loc.dim()-1))
-        return Xu, self.kernel, u_loc, u_scale_tril
+        return self.kernel, Xu, self.mean_function, u_loc, u_scale_tril
```

Comment:
It is generally error-prone to pack and unpack long tuples. Is there a good reason to return `self.kernel` and `self.mean_function` here? The only use I see is in `self.forward()`, where we could instead use `self.*` directly.

Reply:
Yes, we can use `self.kernel`, `self.mean_function` in forward. I made a return for them from `guide` to illustrate that they are ingredients to compute prediction. I agree that they are unnecessary so will remove soon. The reason to call `guide` in `forward` is to set `guide` mode for model/kernel and get parameters for prediction.